modules/op-configuring-ssh-authentication-for-git.adoc

```diff
@@ -0,0 +1,97 @@
+// This module is included in the following assembly:
+//
+// *openshift-docs/cicd/pipelines/authenticating-pipelines-using-git-secret.adoc
+
+[id="op-configuring-ssh-authentication-for-git_{context}"]
+= Configuring SSH authentication for Git
+
+For SSH authentication, {pipelines-shortname} creates a `~/.ssh/config` file containing the SSH private key specified in the secret, before executing any steps in a task run or a pipeline run. When the steps execute, {pipelines-shortname} uses the key to retrieve the pipeline resources specified in a run.
+
+[NOTE]
+====
+It is recommended to use SSH based authentication over basic authentication.
+====
+
+[discrete]
+.Procedure
+
+To configure a Git secret of the `ssh-auth` type:
+
+. Generate a link:https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent[SSH private key], or copy an existing private key which is usually available in the `~/.ssh/id_rsa` file.
+. In the `secret.yaml` file, specify the username and password required to access the target Git repository.
++
+[source,yaml,subs="attributes+"]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  name: ssh-key
+  annotations:
+    tekton.dev/git-0: github.com
+type: kubernetes.io/ssh-auth
+stringData:
+  ssh-privatekey: <private-key>
+  known_hosts: <known-hosts>
+----
++
+Set the value of the `known_hosts` field to the generated `ssh-privatekey` value.
++
+[CAUTION]
+====
+If you omit the private key, {pipelines-shortname} accepts the public key of any server.
+====
++
+[NOTE]
+====
+You can specify a custom SSH port in the secret by adding `:<port>` at the end of the domain hosting the Git repository. For example, to connect to the 2222 port of the `github.com`, you can modify the annotation as `tekton.dev/git-0: github.com:2222`.
+====
++
+. In the `serviceaccount.yaml` file, associate the secret with the appropriate service account.
```

Comment:
As a user reading the following example, where do I get the value that you show as "ssh-key" from? Is it a file name? 

In the instruction, you mention a "service account." Is "build-bot" the service account name? How do I know what the *appropriate* service account is? Can you be more specific?

Reply:
If the following code block contains example values, maybe end the preceding step with " For example:" 
Otherwise, consider using angle brackets to identify which values are user-replaceable and describe what the values should be. For example, `<ssh-key filename>`.